dcli/language/types_helper.py

```diff
@@ -0,0 +1,368 @@
+import json
+
+import inquirer
+
+from dcli.generator import file
+
+PRIMITIVE_TYPE = 'Primitive type (eg int, character)'
+
+DEFAULT_VALUE = 'defaultValue'
+
+METHOD_NAME = 'METHOD_NAME'
+
+FILE_NAME = 'FILE_NAME'
+
+GENERIC = 'Generic (eg T, R)'
+
+ARRAY_DIMENSION = 'number'
+
+TYPE_NAME = 'name'
+
+CLASS_NAME = 'className'
+
+ARRAY = 'Array'
+
+CONTINUE = 'continue'
+
+YES = 'Yes'
+
+NO = 'No'
+
+PARAM_NAME = 'paramName'
+
+TYPE = 'type'
+
+PARAMETRIZED_TYPE = 'Type to parametrize (eg a list)'
+
+KNOWN_LIB_LANGUAGE_TYPE = 'Existing type from language lib (eg InputStream in java)'
+OWN_TYPE = 'Your own type (will be created at generation)'
+
+
+def does_user_continue(parameter_="Do you want to continue"):
+    continue_questions = [
+        inquirer.List(CONTINUE,
+                      message=parameter_,
+                      choices=[YES, NO],
+                      ),
+    ]
+    answer = inquirer.prompt(continue_questions)
+    return answer[CONTINUE] == YES
+
+
+def select_type_from_dict(question_msg='What type do you need ?', types_dict=[]):
+    questions = [
+        inquirer.List(TYPE,
+                      message=question_msg,
+                      choices=types_dict,
+                      ),
+    ]
+    answers = inquirer.prompt(questions)
+    return answers
+
+
+def get_param_name():
+    arg_questions = [
+        inquirer.Text(PARAM_NAME, message="What is the parameter name ?")
+    ]
+    return inquirer.prompt(arg_questions)[PARAM_NAME]
+
+
+def read_json_types(json_file):
+    loaded_types = json.load(json_file)
+    types_to_choose = {}
+    for current_type in loaded_types:
+        name_ = current_type[TYPE_NAME]
+        default_value_ = current_type[DEFAULT_VALUE]
+        my_val = StructDescription(type_name=name_, default_value=default_value_)
+
+        types_to_choose.update({my_val.type_name: my_val.default_value})
+    return types_to_choose
+
+
+class FormAnswersCollector:
+    def __init__(self, language_id,
+                 filename,
+                 allow_typing=True,
+                 have_generics=True,
+                 have_parametrized=True,
+                 allow_type_creation=True,
+                 allow_lib=True):
+        self.allow_typing = allow_typing
+        json_file = file.openFile(language_id, 'resources' + '/' + filename)
+        self.types_to_choose_dict = read_json_types(json_file)
+        self.structure_holder = self.get_datatype_from_user(
+            language_id=language_id,
+            filename=filename,
+            have_generics=have_generics,
+            have_parametrized=have_parametrized,
+            allow_type_creation=allow_type_creation,
+            allow_lib=allow_lib)
+
+    def get_datatype_from_user(self, language_id,
+                               filename,
+                               have_generics=True,
```

Comment:
Missing spaces between =, same within the all file

Reply:
default value affectation into python method parameters shouln't have spaces